NoNewGlobals for Comm::AcceptLimiter::Instance_

kinkie opened the pull request:
Inspired by Coverity. CID 1441988: Useless call (USELESS_CALL).

yadij commented on src/comm.cc:
```diff
@@ -1155,7 +1155,7 @@ comm_init(void)
 
     /* make sure the accept() socket FIFO delay queue exists */
-    Comm::AcceptLimiter::Instance();
+    (void)Comm::AcceptLimiter::Instance();
```
I agree with Alex on this one, please implement conversion to "No new globals" design.

kinkie commented:
> Thank you for improving this code and addressing Coverity defect at the same time!
> 
> I have adjusted PR title to better (IMO) reflect current PR state. I have also removed a technical problem and a misspelling from PR description. Please adjust further as needed.

I am confused by the change. Past PRs doing similar things (e.g #1750, #1781 , #1791, #1766, and more) used the NoNewGlobals identifier. Why is this PR not following the same convention?

rousskov commented:
> Past PRs doing similar things used the NoNewGlobals identifier. Why is this PR not following the same convention?

My mistake: When fixing PR title and looking for similar past PRs, I failed to find those NoNewGlobals commits. I should have paid more attention to insufficient number of matches in my search. Sorry.

Should be fixed now. Thank you for spotting this and trying to keep PR metadata up-to-date throughout changes.
